src/fastly_bouncer/main.py

```diff
@@ -0,0 +1,255 @@
+import argparse
+import json
+import logging
+from logging.handlers import RotatingFileHandler
+import os
+import sys
+import signal
+from pathlib import Path
+from math import ceil
+from time import sleep
+from typing import List
+from multiprocessing.pool import ThreadPool
+from threading import Thread
+from importlib.metadata import version
+
+from pycrowdsec.client import StreamClient
+
+
+from fastly_bouncer.fastly_api import ACL_CAPACITY, FastlyAPI
+from fastly_bouncer.service import ACLCollection, Service
+from fastly_bouncer.utils import (
+    with_suffix,
+    SUPPORTED_ACTIONS,
+    get_default_logger,
+    CustomFormatter,
+)
+from fastly_bouncer.config import (
+    Config,
+    ConfigGenerator,
+    FastlyAccountConfig,
+    FastlyServiceConfig,
+    parse_config_file,
+    print_config,
+)
+
+
+VERSION = version("crowdsec-fastly-bouncer")
+
+acl_collections: List[ACLCollection] = []
+services: List[Service] = []
+
+logger: logging.Logger = get_default_logger()
+
+exiting = False
+
+
+def sigterm_signal_handler(signum, frame):
+    global exiting
+    exiting = True
+    logger.info("exiting")
+
+
+signal.signal(signal.SIGTERM, sigterm_signal_handler)
+signal.signal(signal.SIGINT, sigterm_signal_handler)
+
+
+# TODO Avoid nested functions by using starmap
+def setup_fastly_infra(config: Config, cleanup_mode):
+    if Path(config.cache_path).exists():
+        logger.info("cache file exists")
+        with open(config.cache_path) as f:
+            s = f.read()
+
+            if not s:
+                logger.warning(f"cache file at {config.cache_path} is empty")
+            else:
+                cache = json.loads(s)
+                services.extend(list(map(Service.from_jsonable_dict, cache)))
+                logger.info(f"loaded exisitng infra using cache")
+                if not cleanup_mode:
+                    return
+
+    if cleanup_mode:
+        logger.info("clearing fastly infra")
```

Comment:
`clearing` may scare an user (clear all the existing infra). Maybe `cleaning is better`

Reply:
also I would have said `cleaning fastly services` and same on other logs that mention `infra`

Reply:
done